core/src/main/java/io/grpc/ChannelLogger.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+/**
+ * A Channel-specific logger provided by GRPC library to {@link LoadBalancer} implementations.
+ * Information logged here goes to <string>Channelz</strong>, and to Java logger as well.
+ */
+public abstract class ChannelLogger {
+  /**
+   * Log levels.  See the table below for the mapping from the ChannelLogger levels to Channelz
+   * severity level (see {@code ChannelTraceEvent} from <a
+   * href="https://github.com/grpc/grpc-java/blob/master/services/src/main/proto/grpc/channelz/v1/channelz.proto">channelz.proto</a>)
+   * and Java logger levels.  Note that {@code DEBUG} level is not recorded on Channelz.
+   * <pre>
+   * +---------------------+-------------------+-------------------+
+   * | ChannelLogger Level | Channelz Severity | Java Logger Level |
+   * +---------------------+-------------------+-------------------+
+   * | DEBUG               | N/A               | FINEST            |
```

Comment:
This table should be encoded as arguments to the enum valuies:


```java
public enum Level {
  DEBUG(null, "CT_INFO", Level.FINEST),
}
```

That way the mapping can be without a switch statement below.

Reply:
It's suffiicient to make this a String instead of a hard ref, and then have a test that assert they all work.

Reply:
For these fields to be accessible from `ChannelLoggerImpl`, they would need to be public, which technically exposes these `CT_` enums even not referencing them directly. Doesn't seem enough benefit from doing that.

Reply:
These could be exposed via an internal accessor.  

The benefit is that we can use hard references to see exactly how they map, rather than relying on switch statements.